pkg/template/cue.go

```diff
@@ -59,7 +63,7 @@ func (c *cue) template(dirPath string, input io.Reader) exec.CmdRunResult {
 		args = append(args, c.opts.Paths...)
 	}
 
-	vals := Values{c.opts.ValuesFrom, c.appContext, c.coreClient}
+	vals := Values{c.opts.ValuesFrom, c.appContext, c.coreClient, c.deployFactory, c.log}
```

Comment:
this is actually a good example of dep-inject + factory. right now newCue has the responsbility of creating Values{} -- because of that Value's dependencies are leaking into newCue (e.g. coreClient and deployFactory). what i would recommend doing is newCue to take a ValuesFactory. ValuesFactory would accept coreClient and deployFactory and hide all that from cue's usage: `vals := c.valuesFactory.NewValues(c.opts.ValuesFrom, c.appContext)`